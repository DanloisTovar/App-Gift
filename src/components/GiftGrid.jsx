import React from 'react';

// !importar custom hooks:
import { useFetchGifts } from '../hooks';

// !importar componentes:
import { GiftItem } from './GiftItem';

export const GiftGrid = ({ categoria }) => {
    // !hooks:
    const { newImages, isLoading } = useFetchGifts(categoria);

    return (
        <>
            <h3>{categoria}</h3>
            {isLoading && (
                <p className="animate__animated animate__flash">Loading...</p>
            )}

            <div className="card-grid">
                {newImages.map((imagen) => (
                    <GiftItem key={imagen.id} {...imagen} />
                ))}
            </div>
        </>
    );
};
