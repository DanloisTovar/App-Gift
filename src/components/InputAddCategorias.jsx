import React from 'react';
import PropTypes from 'prop-types';
// !hooks:
import { useState } from 'react';

export const InputAddCategorias = ({ onAddCategorias }) => {
    // hooks:
    const [inputValue, setInputValue] = useState('');

    // !Funciones:
    const handleOnchange = ({ target }) => {
        // desestructura el event para que solo traiga el target " { target } "!!
        setInputValue(target.value);
    };

    const handleOnSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;

        onAddCategorias(inputValue.trim());

        setInputValue('');
    };

    return (
        <form onSubmit={handleOnSubmit}>
            <br></br>
            <input
                type="text"
                value={inputValue}
                onChange={handleOnchange}
                onSubmit={handleOnSubmit}
                placeholder="Agregue la categorÃ­a a buscar..."
            />
            <br></br>
        </form>
    );
};

InputAddCategorias.propTypes = {
    onAddCategorias: PropTypes.func.isRequired,
};
