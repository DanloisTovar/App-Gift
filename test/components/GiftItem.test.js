import { render, screen } from '@testing-library/react';
import { GiftItem } from '../../src/components/GiftItem';

describe('Prueba de funcionamiento jest', () => {
    test(' verificar PropTypes y hacer snapshot ', () => {
        const title = 'Hola Mundo';
        const url = 'https://localhost:3000';
        const { container } = render(<GiftItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostar la imagen con el URL y el ALT indicado', () => {
        const title = 'Hola Mundo';
        const url = 'https://localhost:3000/';

        render(<GiftItem title={title} url={url} />);
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(src);
        expect(alt).toBe(alt);
    });

    test('Debe de mostrar el titulo en el componente ', () => {
        const title = 'Hola Mundo';
        const url = 'https://localhost:3000/';

        render(<GiftItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    });
});
