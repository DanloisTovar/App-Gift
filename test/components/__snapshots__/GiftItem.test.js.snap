// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de funcionamiento jest  verificar PropTypes y hacer snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola Mundo"
      src="https://localhost:3000"
    />
    <p>
      Hola Mundo
    </p>
  </div>
</div>
`;
